Update CreateCampaignRequest so that we can override the default MessageChannelKinds of the specified message template

spapoutsis opened the pull request.

Commit:
#45374

add missing json converter that is required for GuidOrAlias custom type

Commit:
#45374

messageTemplateChannel(s)

Commit:
#45374

GuidOrAlias in template Api

cleftheris commented on src/Indice.Features.Messages.Core/Models/Requests/CampaignRequestBase.cs:
```diff
@@ -25,4 +25,6 @@ public class CampaignRequestBase
     /// <summary>Optional data for the campaign.</summary>
     public dynamic? Data { get; set; }
+    /// <summary>If exists, it overrides the Channels collection of the specified template. </summary>
+    public List<MessageChannelKind>? MessageTemplateChannels { get; set; }
```
Since we are using flags Enums everywhere would you consider this property being a plain 

`public MessageChannelKind? MessageTemplateChannels { get; set; }` 

or would this be too much of a hustle?

This would not affect generated clients whatsoever since there are available converters both for the openapi spec and the model binder/serializer that will handle it as an array for the outside world

spapoutsis replied:
sure! let me try it and send another commit

spapoutsis replied:
seems ok now, please check again!

cleftheris committed:
Merge branch 'develop' into feature/US-45370/WI-45374

Commit:
#45374

updated contract based on PR's comments

Commit:
Merge branch 'develop' into feature/US-45370/WI-45374

cleftheris committed:
Enhance nullable enum handling in JsonStringArrayEnumFlagsConverter

Updated the JsonStringArrayEnumFlagsConverter<TEnum> to better handle nullable enums by retrieving the underlying type during string parsing. Adjusted array end handling for correct enum value parsing.

Added a new test case in JsonStringArrayEnumFlagsConverterTests to validate deserialization of comma-separated enum values, ensuring proper interpretation and combination into enum flags.

cleftheris committed:
Simplify initialization from a `List<MessageChannelKind>()` to array syntax `[]`.

Commit:
#45374

Commit:
#45374

removing debug entry

cleftheris reviewed (approved):
Ship it 🎉

Commit:
Merge branch 'develop' into feature/US-45370/WI-45374
